### PR TITLE
Fix loading lightmaps from gltf / glb meshes

### PR DESCRIPTION
**Pull request by iche033**
# 🦟 Bug fix

## Summary

Fix loading gltf / glb mesh files with lightmaps. Needed by the Ionic demo world.

Currently the lightmap is ignored if there's a combined roughness + metalness texture. This PR fixes the problem by loading the lightmap (ambient occlusion data) from the roughness + metalness texture.

Tested with a few [gltf sample models](https://github.com/KhronosGroup/glTF-Sample-Models/tree/main/2.0) (water bottle, barramundi fish, damaged helmet)

Here are screenshots of the [BarramundiFish](https://github.com/KhronosGroup/glTF-Sample-Models/tree/main/2.0/BarramundiFish) before (left) and after (right) the change. I removed the albedo map to better visualize the effect of the lightmap:

barramundi_no_lightmap

barramundi_lightmap

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: graphics/src/AssimpLoader.cc [hunk 1]
```diff
@@ -462,22 +462,50 @@ MaterialPtr AssimpLoader::Implementation::CreateMaterial(
...
       pbr.SetLightMap(texName, uvIdx, texData);
     }
+    // else split the occlusion data from the metallicRoughness texture
+    else
+    {
+      // R channel contains the occlusion data
+      // Note we are still creating an RGBA texture which seems watesful
+      // but that's what gz-rendering expects
+      auto origRGBAData = texData->RGBAData();
+      std::vector<unsigned char> texRData(origRGBAData.size());
```
Comment (ahcorde) [inline graphics/src/AssimpLoader.cc:495]: include `<vector>`

File: graphics/src/AssimpLoader.cc [hunk 2]
```diff
@@ -462,22 +462,50 @@ MaterialPtr AssimpLoader::Implementation::CreateMaterial(
...
+      for (unsigned int i = 0; i < origRGBAData.size(); i+=4)
+      {
+        auto r = origRGBAData.at(i);
+        texRData[i] = r;
+        texRData[i + 1] = r;
+        texRData[i + 2] = r;
+        texRData[i + 3] = 255;
+      }
+      auto tex = std::make_shared<Image>();
```
Comment (ahcorde) [inline graphics/src/AssimpLoader.cc:504]: include `<memory>`

**Review by ahcorde — CHANGES_REQUESTED** [review]
just some nits, otherwise LGTM

File: graphics/src/AssimpLoader.cc (hunk 2 above)
Comment (iche033) [reply graphics/src/AssimpLoader.cc:506]: [29b17f1](https://github.com/gazebosim/gz-common/pull/630/commits/29b17f164e5bc7da886db0c00bc44d04f10baa21)

File: graphics/src/AssimpLoader.cc (hunk 1 above)
Comment (iche033) [reply graphics/src/AssimpLoader.cc:497]: [29b17f1](https://github.com/gazebosim/gz-common/pull/630/commits/29b17f164e5bc7da886db0c00bc44d04f10baa21)

**Comment by iche033** [comment]
Some tests in UNIT_AssimpLoader_TEST on hombrew are failing. I noticed that failing in main as well and could be due to a new AssimpLoader homebrew bottle. I ticketed an issue for it: https://github.com/gazebosim/gz-common/issues/633

**Review by ahcorde — APPROVED** [review]

**Comment by iche033** [comment]
Fixed homebrew CI in https://github.com/gazebosim/gz-common/pull/634

**Pull request merged**